util/algorithm_identifier_test.go

```diff
@@ -131,3 +137,44 @@ func TestCheckAlgorithmIDParamNotNULL(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPublicKeyOID(t *testing.T) {
+
+	path := "../testdata/rsaSigAlgoNoNULLParam.pem"
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		panic(fmt.Sprintf(
+			"Unable to read test certificate from %q - %q "+
+				"Does a unit test have an incorrect test file name?\n",
+			path, err))
+	}
+
+	if strings.Contains(string(data), "-BEGIN CERTIFICATE-") {
+		block, _ := pem.Decode(data)
+		if block == nil {
+			panic(fmt.Sprintf(
+				"Failed to PEM decode test certificate from %q - "+
+					"Does a unit test have a buggy test cert file?\n",
+				path))
+		}
+		data = block.Bytes
+	}
+
+	cert, err := x509.ParseCertificate(data)
+	if err != nil {
+		panic(fmt.Sprintf(
+			"Failed to parse x509 test certificate from %q - "+
+				"Does a unit test have a buggy test cert file?\n",
+			path))
+	}
+
```

Comment:
Please use `test.ReadTestCert` for this. That will avoid duplicating the file read, pem decode, and cert parse as well as avoiding the need to specify the `../testdata/` path component here.

Reply:
Doing this produces an "import cycle not allowed in test". How should we handle this better?

Reply:
Ahh darn. You're right. `util` code can't import `test` because that package imports `lint` which imports `util`. :dizzy_face: 

I think given this problem & the discussion we've had about the `TestGetPublicKeyOID` function I would vote that we remove that test case and the helper function entirely. We get equivalent coverage of the core logic without code duplication/import cycles via the unit tests for the overall id-RSASSA-PSS lint.

Does that work for you?

Reply:
Yes.